### PR TITLE
Make sure the trying timer is not running when deleting a UASTsx. 

### PR DESCRIPTION
**Pull request by AlexHockey**
Hi Rob, please can you review my fix for #1325? This builds on your earlier PR (#1344) which I've already reviewed, so it probably makes sense for you to review my final commit. 

I have checked that:

* The UTs all pass successfully. 
* The live tests work and the error log is not produced.

### CODE REVIEW

File: src/basicproxy.cpp [hunk 1]
```diff
@@ -438,19 +438,41 @@ BasicProxy::UASTsx::UASTsx(BasicProxy* proxy) :
...
+
+  if (_tsx != NULL)
+  {
+    // Unbind from the PJSIP transaction.
+    _proxy->unbind_transaction(_tsx);
+    _tsx = NULL;
+
+    // The trying timer should be running when we have a PJSIP transaction, so
+    // cancel it as well.
+    cancel_trying_timer();
```
Comment (rkd-msw) [inline src/basicproxy.cpp]: As this is the only place where `cancel_trying_timer` is called, can we just inline it? I'm nervous about someone calling it from some other place and unwittingly violating our safety invariant.

If you'd like to keep it as a separate function, can we stick `CHECK_PJ_TRANSPORT_THREAD();` in there too/instead?

File: include/basicproxy.h [hunk 2]
```diff
@@ -211,6 +211,10 @@ class BasicProxy
     /// and vice-versa.  This must be called before destroying either transaction.
     void dissociate(UACTsx* uac_tsx);
 
+    /// Unbind this UASTSx from the PJSIP transaction (if any) and cancel any
+    /// timers that should only be running when the UASTsx is bound.
+    void unbind_from_pjsip_tsx();
```
Comment (rkd-msw) [inline include/basicproxy.h]: "unbind" doesn't feel like the best name to me, but I can't think of a better one.

**Comment by rkd-msw** [comment]
Looks good.

File: src/basicproxy.cpp (hunk 1 above)
Comment (AlexHockey) [inline src/basicproxy.cpp]: I like the idea of inlining, especially as the code to start the timer is also inlined.

**Comment by AlexHockey** [comment]
I'm not merging this into release-94-fixes, so I'm going to close the PR and make a new one into master.

**Pull request closed without merging**